src/symbol/symbol_layout.js

```diff
@@ -713,6 +714,27 @@ function addSymbol(bucket: SymbolBucket,
     const verticalIconBoxStartIndex = verticalIconCollisionFeature ? verticalIconCollisionFeature.boxStartIndex : bucket.collisionBoxArray.length;
     const verticalIconBoxEndIndex = verticalIconCollisionFeature ? verticalIconCollisionFeature.boxEndIndex : bucket.collisionBoxArray.length;
 
+    // Check if runtime collision circles should be used for any of the collision features.
+    // It is enough to choose the tallest feature shape as circles are always placed on a line.
+    // All measurements are in glyph metrics and later converted into pixels using proper font size "layoutTextSize"
+    let collisionCircleDiameter = -1;
+
+    const getCollisionCircleHeight = (feature: ?CollisionFeature, prevHeight: number): number => {
+        if (feature && feature.circleDiameter)
+            return Math.max(feature.circleDiameter, prevHeight);
+        return prevHeight;
+    };
+
+    collisionCircleDiameter = getCollisionCircleHeight(textCollisionFeature, collisionCircleDiameter);
+    collisionCircleDiameter = getCollisionCircleHeight(verticalTextCollisionFeature, collisionCircleDiameter);
+    collisionCircleDiameter = getCollisionCircleHeight(iconCollisionFeature, collisionCircleDiameter);
+    collisionCircleDiameter = getCollisionCircleHeight(verticalIconCollisionFeature, collisionCircleDiameter);
+    const useRuntimeCollisionCircles = (collisionCircleDiameter > -1) ? 1 : 0;
```

Comment:
true / false?

Reply:
type conversion from bool to int is required because the variable is serialized as a number